prov/sm2/src/sm2_progress.c

```diff
@@ -267,12 +407,42 @@ static int sm2_alloc_xfer_entry_ctx(struct sm2_ep *ep,
 static int sm2_progress_recv_msg(struct sm2_ep *ep,
 				 struct sm2_xfer_entry *xfer_entry)
 {
+	struct sm2_av *av =
+		container_of(ep->util_ep.av, struct sm2_av, util_av);
+	struct sm2_mmap *map = &av->mmap;
 	struct fid_peer_srx *peer_srx = sm2_get_peer_srx(ep);
 	struct fi_peer_rx_entry *rx_entry;
 	struct sm2_av *sm2_av;
 	fi_addr_t addr;
 	int ret;
 
+	/* sm2_proto_ipc_host_to_dev protocol means that the sender received an
+	 * IPC handle from the receiver. So the sender needs to do a CUDA memcpy
```

Comment:
Personally, I would drop the comment entirely since that's the definition of this protocol. If an explanation needs to go anywhere, it should just be where the protocol is defined/in the header.